cwbi-auth-http-client/src/main/java/hec/army/usace/hec/cwbi/auth/http/client/TokenUrlDiscoveryService.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2025 Hydrologic Engineering Center
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package hec.army.usace.hec.cwbi.auth.http.client;
+
+import java.util.Objects;
+import mil.army.usace.hec.cwms.http.client.ApiConnectionInfo;
+
+public abstract class TokenUrlDiscoveryService {
```

Comment:
I don't see any need for an abstract class here, an interface makes more sense. I haven't seen your implementation, but it seems to make sense to me that you'd have an interface with just a `ApiConnectionInfo getTokenUrl()`.

Also, I don't understand the reason for parameterizing the ApiConnectionInfo into `retrieveTokenUrl` rather than making the `getApiConnectionInfo()` itself protected (you're already testing it, which suggests it should be public).

Reply:
Yeah, originally there was some async business, but i removed that. This can just be an interface. I think this can just be `String discoverTokenUrl()`

Reply:
String means that the calling API needs to know how to setup the SSL pieces, cookies, etc. That seems like parts that belong as part of the implementation.

Reply:
the calling API doesnt need to set up the SSL pieces, just needs to implement the query to get back the token url string.
Here is an example in the cumulus client of that implementation:

https://github.com/HydrologicEngineeringCenter/cumulus-client/blob/CWMS-2055_dynamic_keycloak_discovery/cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/auth/CumulusTokenUrlDiscoveryService.java